astroid/interpreter/_import/spec.py

```diff
@@ -423,7 +425,19 @@ def _find_spec_with_path(
     raise ImportError(f"No module named {'.'.join(module_parts)}")
 
 
+@dataclass(frozen=True)
+class SpecArgs:
+    key: tuple
+    modpath: list[str] = field(compare=False, hash=False)
+    path: Sequence[str] | None = field(compare=False, hash=False)
```

Comment:
I'm concerned about not adding `path` to the cache key. I did see this:

> I don't think search paths should be cached? If I already know where the package is found, does it matter if you give me different search paths? This point I am not so sure about, but it was my logic.

But this assumes the package is found. If the package is not found with a given search path, but then re-attempted with a better path, we want the search to run again.

Reply:
I don't think results are cached if an exception is raised? From what I see, `_find_spec_with_path` raises an ImportError if it failed to find the requested module. That would cover this case :thinking: